lib/srv/monitor_test.go

```diff
@@ -57,6 +58,86 @@ func newTestMonitor(ctx context.Context, t *testing.T, asrv *auth.TestAuthServer
 	return conn, emitter, cfg
 }
 
+func TestConnectionMonitorLockInForce(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	asrv, err := auth.NewTestAuthServer(auth.TestAuthServerConfig{
+		Dir:   t.TempDir(),
+		Clock: clockwork.NewFakeClock(),
+	})
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, asrv.Close()) })
+
+	// Create a connection monitor that points to our test
+	// Auth server.
+	emitter := eventstest.NewChannelEmitter(1)
+	monitor, err := NewConnectionMonitor(ConnectionMonitorConfig{
+		AccessPoint: asrv.AuthServer,
+		Emitter:     emitter,
+		Clock:       asrv.Clock(),
+		Logger:      logrus.StandardLogger(),
+		LockWatcher: asrv.LockWatcher,
+		ServerID:    "test",
+	})
+	require.NoError(t, err)
+
+	lock, err := types.NewLock("test-lock", types.LockSpecV2{Target: types.LockTarget{User: "test-user"}})
+	require.NoError(t, err)
+
+	identity := &authz.LocalUser{
+		Username: "test-user",
+		Identity: tlsca.Identity{
+			Username: "test-user",
+		},
+	}
+
+	authzCtx := &authz.Context{
+		Checker:          mockChecker{},
+		Identity:         identity,
+		UnmappedIdentity: identity,
+	}
+
+	t.Run("lock created after connection has been established", func(t *testing.T) {
+		// Create a fake connection and monitor it.
+		conn := &mockTrackingConn{closedC: make(chan struct{})}
+		monitorCtx, err := monitor.MonitorConn(ctx, authzCtx, conn)
+		require.NoError(t, err)
+		require.Nil(t, monitorCtx.Err())
```

Comment:
Nit: Should we assert that after the locks creation that this context is cancelled as well as the connection being closed?